### ISSUE TITLE
Papers in the future

### ISSUE THREAD

**Issue by Phyks** [dissemin/dissemin#643]
We currently have a couple of issues with dates.

For instance, https://dissem.in/p/93660579/determinacion-del-valor-pronostico-a-6-meses-del-puntaje-de-riesgo-oesil-en-una-cohorte-colombiana-con-sincope-evaluada-en-el-servicio-de-urgencias-primera-experiencia-latinoamericana is labelled in 2099 whereas it is published in 2017.

Seems that we have about 1k such papers in the future https://dissem.in/search/?pub_after=2019. Not sure where is this coming from…

**Comment by wetneb** [comment]
I think this is generally due to publisher metadata errors indeed. I suspect publishers put down a dummy date when first depositing the metadata, and then update it once they know the actual publication date (or when they figure out that this is an issue). In this particular example, the DOI metadata looks fine:
http://doi-cache.dissem.in/10.1016/j.acmx.2017.06.007

To solve this, we need to:
- [ ] make sure ingesting the sources updates the OaiRecord publication dates (`OaiRecord.pubdate`)
- [ ] make sure `Paper.update_availability` refreshes the date on the Paper object based on the `OaiRecord.pubdate`s

**Comment by wetneb** [comment]
We should also note that some publishers purposely expose papers "in the future" to inflate their impact factor, so publication dates up to a few months/years in the future are "acceptable" somehow.

Because of that, it seems that we already have a 3 year cutoff date for papers in the future, so the remaining ones are probably leftover from an earlier import where this was not implemented.
https://github.com/dissemin/dissemin/blob/edcddc704b8fbabf2564c06aeabf661f741b24ec/papers/utils.py#L539

**Comment by a3nm** [comment]
I think we have more than 12 papers which are >3 years into the future, no? See https://dissem.in/search/?authors=

These papers are quite easy to see as they are the first ones that people stumble onto if they submit the search form on the main page with nothing in it.

**Comment by Phyks** [comment]
> I think we have more than 12 papers which are >3 years into the future, no? See https://dissem.in/search/?authors=

Indeed, sounds like we recently found new ones (maybe with the index update?). Or I completely failed in my latest message :/

Anyways,

> These papers are quite easy to see as they are the first ones that people stumble onto if they submit the search form on the main page with nothing in it.

:+1: and that's why I consider it something important to fix :)
